src/azure-cli/azure/cli/__main__.py

```diff
@@ -40,6 +40,9 @@ def cli_main(cli, args):
 
     exit_code = cli_main(az_cli, sys.argv[1:])
 
+    if az_cli.result.error is not None and not telemetry.has_exceptions():
+        telemetry.set_exception(az_cli.result.error, fault_type='')
+
     if exit_code and exit_code != 0:
         telemetry.set_failure()
```

Comment:
Can we resuse this part of code?

Reply:
Yes. You can call `telemetry.set_exception()` with a self-defined fault_type in the module code. Here we check that if you have an error but have't added it to telemetry, then we add it when the command finishes executing, but we cannot decide the fault_type.

Reply:
I was wonder why `telemetry.has_exceptions` and `az_cli.result.exit_code != 0` occur in different timming.

As far as I understand, when `az_cli.result.error is not None` which also means `az_cli.result.exit_code != 0`. 
At this time, `telmetry.has_exceptions` and `telemetry.set_failure` should put together under this condition.

Which means, should put `telemetry.set_exception(az_cli.result.error, fault_type='')` under the if statement `if exit_code and exit_code != 0:` as `telemetry.set_failure()`.

Reply:
I moved the new code under `if exit_code and exit_code != 0:` in [knack](https://github.com/microsoft/knack/blob/master/knack/util.py#L16-L17), but since `exit_code` and `error` are set differently, I would keep checking the error to be robust.